framework/meta/src/template.rs

```diff
@@ -48,3 +57,60 @@ pub fn copy_template_to_location(template: &str, location: &Path) {
         .join(template);
     let _ = copy_dir(contract_path, location);
 }
+
+pub fn update_dependencies(template: &str) {
+    update_dependencies_root(template);
+    update_dependencies_wasm(template);
+    update_dependencies_meta(template);
+}
+
+pub fn update_dependencies_root(template: &str) {
+    let path_buf = get_canonicalized_path(template, ROOT_CARGO_TOML);
+    let root_cargo_toml_path = Path::new(&path_buf);
+    let mut toml = CargoTomlContents::load_from_file(root_cargo_toml_path);
+
+    toml.dev_dependencies_mut().clear();
+
+    remove_paths_from_dependencies(&mut toml, root_cargo_toml_path, &[]);
+}
+
+pub fn update_dependencies_meta(template: &str) {
+    let path_buf = get_canonicalized_path(template, META_CARGO_TOML);
+    let root_cargo_toml_path = Path::new(&path_buf);
+    let mut toml = CargoTomlContents::load_from_file(root_cargo_toml_path);
+
+    remove_paths_from_dependencies(&mut toml, root_cargo_toml_path, &[template]);
+}
+
+pub fn update_dependencies_wasm(template: &str) {
+    let path_buf = get_canonicalized_path(template, WASM_CARGO_TOML);
+    let root_cargo_toml_path = Path::new(&path_buf);
+    let mut toml = CargoTomlContents::load_from_file(root_cargo_toml_path);
+
+    remove_paths_from_dependencies(&mut toml, root_cargo_toml_path, &[template]);
+}
+
+pub fn get_canonicalized_path(template: &str, toml: &str) -> PathBuf {
+    Path::new(template)
+        .join(toml)
+        .canonicalize()
+        .unwrap_or_else(|err| {
+            panic!("error canonicalizing input path: {err}",);
+        })
+}
+
+pub fn remove_paths_from_dependencies(
+    cargo_toml_contents: &mut CargoTomlContents,
+    path: &Path,
+    ignore_deps: &[&str],
+) {
+    let deps_map = cargo_toml_contents.dependencies_mut();
```

Comment:
You can iterate here through both dependencies and dev-dependencies.

Either chain the iterators, or 2 successive for loops.